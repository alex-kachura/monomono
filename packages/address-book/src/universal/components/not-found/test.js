import React from 'react';
import NotFoundPage from '.';
import { render, cleanup } from 'react-testing-library';
import { AppProvider } from '@oneaccount/react-foundations';
import { DefaultThemeProvider } from '@beans/theme';

describe('NotFound component', () => {
  afterEach(cleanup);

  function renderComponent() {
    return render(
      <AppProvider
        appConfig={{
          getLocalePhrase: (key) => key,
          config: {
            GB: {
              externalApps: {
                tescoHomepage: '/mock-path',
              },
            },
          },
          region:'GB',
        }}
      >
        <DefaultThemeProvider>
          <NotFoundPage />
        </DefaultThemeProvider>
      </AppProvider>
    );
  }

  it('should render correctly', () => {
    const { asFragment } = renderComponent();
    const fragment = asFragment();

    expect(fragment).toMatchSnapshot();
  });
});
