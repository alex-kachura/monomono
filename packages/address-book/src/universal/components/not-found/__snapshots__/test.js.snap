// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="base-components__BaseElement-sc-150tvch-0 typography___styledElement-sc-186l0ce-0 hmbiuS"
    >
      pages.not-found.heading
    </h1>
    <p
      class="base-components__BaseElement-sc-150tvch-0 typography___styledElement-sc-186l0ce-0 eEYJsv"
    >
      pages.not-found.paragraphs.0
    </p>
    <p
      class="base-components__BaseElement-sc-150tvch-0 typography___styledElement-sc-186l0ce-0 eEYJsv"
    >
      pages.not-found.paragraphs.1
    </p>
    <a
      class="sc-bwzfXH fGCCLo styled__TextButtonLink-ipdqot-0 drVknR"
      href="/mock-path"
    >
      <span
        class="beans-link__text styled__TextSpan-ipdqot-3 oPKYq"
      >
        pages.not-found.home
      </span>
    </a>
  </div>
</DocumentFragment>
`;
